test/e2e/proxy_test.go

```diff
@@ -99,3 +140,63 @@ func (tc *testContext) waitForValidTrustedCAConfigMap() error {
 	}
 	return nil
 }
+
+// getSystemEnvVar returns the value corresponding to the input proxy ENV var as set in the registry
+func (tc *testContext) getSystemEnvVar(addr, variableName string) (map[string]string, error) {
+	command := fmt.Sprintf("Get-ChildItem -Path Env: | Where-Object -Property Name -eq '%s' | Format-List ",
+		variableName)
+	out, err := tc.runPowerShellSSHJob(fmt.Sprintf("get-%s-system-env-var",
+		strings.ToLower(strings.ReplaceAll(variableName, "_", "-"))), command, addr)
+	if err != nil {
+		return nil, fmt.Errorf("error running SSH job: %w", err)
+	}
+	value := parseEnvVars(strings.Split(out, "\n"))
+	return value, nil
+}
+
+// getServiceProxyEnvVars returns a map of all environment variables present in a service's config
+func (tc *testContext) getServiceProxyEnvVars(addr, svcName string) (map[string]string, error) {
+	command := fmt.Sprintf("Get-Process %s | ForEach-Object { $_.StartInfo.EnvironmentVariables.GetEnumerator() "+
+		"| Format-List }",
+		svcName)
+	out, err := tc.runPowerShellSSHJob(fmt.Sprintf("get-%s-service-env-vars",
+		strings.ReplaceAll(svcName, "_", "-")), command, addr)
+	if err != nil {
+		return nil, fmt.Errorf("error running SSH job: %w", err)
+	}
+	return parseEnvVars(strings.Split(out, "\n")), nil
+}
+
+// parseEnvVars parses a given slice of string containing all environment variables with their name, value pairs
+// and returns a map of ENV vars to their corresponding values.
+// Sample input: ["Name  : HTTP_PROXY", "Value : http://dev:d3436c0b817f7ca8e23f7b47be49945d@10.0.1.10:3128/",
+// "Name  : SHELL", "Value : c:\windows\system32\cmd.exe"]
+func parseEnvVars(lines []string) map[string]string {
```

Comment:
rename this to parseWindowsEnvVars please.

Why is this made to take a slice of strings and not the direct output of  the powershell command?